apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts

```diff
@@ -553,6 +584,23 @@ function cleanObject(
   );
 }
 
+function convertEmptyStringsToNull(obj: Record<string, unknown> | undefined): Record<string, unknown> | undefined {
```

Comment:
@djabarovgeorge can you please elaborate why is this necessary?

Reply:
definitely! this is necessary because it allows us to alert the user when they've entered a `nullish` value in the Dashboard. Sending and storing an empty string counts as a valid required string, even though the user has "removed" the text.  
https://www.loom.com/share/6c100325b9a140f6b97fccb0b8fc7991

Another solution could be adding a UI helper component to indicate that the user deleted the value. This way, the Dashboard would send a `nullish` value instead of an empty string, preserving validation.